src/http/one/TeChunkedParser.cc

```diff
@@ -115,55 +116,66 @@ Http::One::TeChunkedParser::parseChunkSize(Http1::Tokenizer &tok)
  * ICAP 'use-original-body=N' extension is supported.
  */
 bool
-Http::One::TeChunkedParser::parseChunkExtension(Http1::Tokenizer &tok, bool skipKnown)
+Http::One::TeChunkedParser::parseChunkExtensions(Tokenizer &tok, const bool skipKnown)
 {
+    while (parseChunkExtension(tok, skipKnown)) {
+        buf_ = tok.remaining(); // got one extension; there may be more
+    }
+
+    if (skipLineTerminatorIfAny(tok)) {
+        buf_ = tok.remaining();
+        parsingStage_ = theChunkSize ? Http1::HTTP_PARSE_CHUNK : Http1::HTTP_PARSE_MIME;
+        return true; // reached the end of extensions (if any)
+    }
+
+    static const CharacterSet NonLF = (CharacterSet::LF).complement().rename("non-LF");
+    if (tok.skipAll(NonLF) && tok.skip('\n'))
+        throw TexcHere("cannot parse chunk extension"); // <garbage> CR*LF
+    return false; // need more data to finish parsing the extension
+}
+
+bool
+Http::One::TeChunkedParser::parseChunkExtension(Tokenizer &tok, const bool skipKnown)
+{
+    if (!ParseBws(tok)) // Bug 4492: IBM_HTTP_Server sends SP after chunk-size
+        return false;
+
+    if (!tok.skip(';'))
+        return false;
+
+    if (!ParseBws(tok)) // Bug 4492: ICAP servers send SP before chunk-ext-name
+        return false;
+
     SBuf ext;
-    SBuf value;
-    while (
-        ParseBws(tok) && // Bug 4492: IBM_HTTP_Server sends SP after chunk-size
-        tok.skip(';') &&
-        ParseBws(tok) && // Bug 4492: ICAP servers send SP before chunk-ext-name
-        tok.prefix(ext, CharacterSet::TCHAR)) { // chunk-ext-name
-
-        // whole value part is optional. if no '=' expect next chunk-ext
-        if (ParseBws(tok) && tok.skip('=') && ParseBws(tok)) {
-
-            if (!skipKnown) {
-                if (ext.cmp("use-original-body",17) == 0 && tok.int64(useOriginBody, 10)) {
-                    debugs(94, 3, "Found chunk extension " << ext << "=" << useOriginBody);
-                    buf_ = tok.remaining(); // parse checkpoint
-                    continue;
-                }
-            }
+    if (!tok.prefix(ext, CharacterSet::TCHAR)) // chunk-ext-name
+        return false;
 
-            debugs(94, 5, "skipping unknown chunk extension " << ext);
+    if (!ParseBws(tok))
+        return false;
 
-            // unknown might have a value token or quoted-string
-            if (tok.quotedStringOrToken(value) && !tok.atEnd()) {
-                buf_ = tok.remaining(); // parse checkpoint
-                continue;
-            }
+    if (tok.skip('=')) {
+        if (!ParseBws(tok))
+            return false;
 
-            // otherwise need more data OR corrupt syntax
-            break;
+        if (!skipKnown) {
+            if (ext.cmp("use-original-body", 17) == 0 && tok.int64(useOriginBody, 10)) {
+                debugs(94, 3, "Found chunk extension " << ext << "=" << useOriginBody);
```

Comment:
Please avoid duplicating the method name and, to optimize, print a single character instead of a string:
```suggestion
                debugs(94, 3, "found " << ext << '=' << useOriginBody);
```

Reply:
Done.